### ISSUE TITLE
SCRAM authentication requires libpq version 10 or above in

### ISSUE THREAD

**Issue by NeilJ247** [brefphp/bref#965]
**Description:**

I am getting the following error when trying to establish a connection with Timeseries (postgres) database in Lambda:

`SCRAM authentication requires libpq version 10 or above in`

My application has to connect to 2 separate databases.  Their versions are:

Timeseries
PostgreSQL 10.15 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.3 20140911 (Red Hat 4.8.3-9), 64-bit

Non-timeseries
PostgreSQL 12.7 (Debian 12.7-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit

**How to reproduce:**

Runtime: provided.al2
Layers: bref:layer.php-74 and bref:layer.console
Extensions: pdo_pgsql
Libraries: Doctrine DBAL

I have a local docker image that I build which extends from php:7.4-fpm-alpine which runs the application fine:

```
FROM php:7.4-fpm-alpine

RUN apk update && \
    apk add -q --no-progress $PHPIZE_DEPS postgresql-dev postgresql-client git && \
    yes | pecl install xdebug pcov && \
    docker-php-ext-install pgsql pdo pdo_pgsql && \
    docker-php-ext-enable pcov xdebug
```

**Comment by t-richard** [comment]
@NeilJ247 if you haven't found an answer to your problem, maybe you could give a try to the `pgsql` layer provided at https://github.com/brefphp/extra-php-extensions

Not sure it would fix the issue but it's worth a try.

**Comment by NeilJ247** [comment]
@t-richard yeah I tried that but it's the same issue.

**Comment by BordelD** [comment]
Same here

**Comment by NeilJ247** [comment]
Corrected the original description.  I had the Postgres versions the wrong way around for the databases.

Our Timeseries DB is now upgraded to 12.9 and I have upgraded to PHP 8 but the issue still persists.

**Comment by NeilJ247** [comment]
Which version of libpq is installed by default on bref?

**Comment by Nyholm** [comment]
Bref is currently running 9.6.24 which was released in November 2021. https://github.com/postgres/postgres/releases/tag/REL9_6_24

I also got this issue when I tried to connect to a new Postgres 14 database.

**Comment by Nyholm** [comment]
I tried to create an extension as @t-richard suggested. But postgres is not a "normal" extension installed with pecl. You have to compile PHP with the postgres extension. 

I solve it by using Postgres server 12.7 instead. I dont know if any version between 12.7 and 14.1 will work.

**Comment by deleugpn** [comment]
If there is any interest, we can try to make a postgres extension layer by installing from remi-repository and then basing it off of #1078

**Comment by MatheusSw** [comment]
Same problem here, really sad that there is no support to a new postgres release.

**Comment by HugoJF** [comment]
I've managed to fix this issue just by upgrading Postgress to 14.2 on https://github.com/HugoJF/bref/commit/0ef754adc5c9207fbdc3da3c0ecfa7465d72d730 but I have absolutely no idea what are the implications of going from 9.6.24 to 14.2.

**Comment by aniamin** [comment]
> 9.6.24

Could anyone connect to postgres 14.2 (RDS) yet? I tried to update libpq5 and libpq-dev but not working yet.
Thanks

**Comment by aniamin** [comment]
> I've managed to fix this issue just by upgrading Postgress to 14.2 on [HugoJF@0ef754a](https://github.com/HugoJF/bref/commit/0ef754adc5c9207fbdc3da3c0ecfa7465d72d730) but I have absolutely no idea what are the implications of going from 9.6.24 to 14.2.

I have tried to connect to Postgresql V14.2 but getting the same error yet. Update libpq5 and liblibpq-dev but not workinmg

**Comment by HugoJF** [comment]
> I have tried to connect to Postgresql V14.2 but getting the same error yet. Update libpq5 and liblibpq-dev but not workinmg

Did you replace both `php`, `php-fpm` and `console` layers? I had to upgrade Postgres and also add `${bref-extra:pgsql-php-80}` to enable the PDO extension.

I was trying to connect to Supabase databases, and that was enough to fix it, even rolled back this upgrade to make sure I would get the SCRAM error, since I was working based on what @MatheusSw reported. I tested it by checking `phpinfo()`, running a simple query and running migrations for my application.
